CenterForOpenScience/cas-overlay
# [Feature] More Informative Login Page for Campaign and Branded Preprints [CAS-1, OSF-7217, PREP-224]

## Tickets

https://openscience.atlassian.net/browse/CAS-1
https://openscience.atlassian.net/browse/OSF-7217
https://openscience.atlassian.net/browse/PREP-224

## Purpose

- CAS login page should contain information about OSF campaigns and branded preprints if user originates from and/or targets those pages.
  - [x] Prereg
  - [x] ERPC
  - [x] Preprints
  - [x] Branded Preprints Phase One (thanks Rusty and Barrett for the logo)

- "Create Account" should redirect user back to targeted campaigns and branded preprints login page
  - [x] "Create Account" link on CAS Login
  - [x] "Create Account" link on Institution Login
  - [x] Update branded preprints register url


### Update for related tickets

- https://openscience.atlassian.net/browse/OSF-7076
  - [x] Update registered services check to display ORCiD login
  - [x] Verify ORCiD link/create confirmation link and destination

- https://openscience.atlassian.net/browse/CAS-3
  - [x] Update with CAS-3 changes
  - [x] Use a data structure to pass campaign the front end
  - [x] Update for proxied campaigns (ember preprints and branded ones)
    - handles proxied campaigns login url:
    ```
    http://localhost:8080/login?service=
        http%3A%2F%2Flocalhost%3A5000%2Flogin%2F%3Fnext%3D
        http%253A%252F%252Flocalhost%253A5000%252Fpreprints%252F
    ```
    and
    ```
     http://localhost:8080/login?service=
        http%3A%2F%2Flocalhost%3A5000%2Flogin%2F%3Fnext%3D
      http%253A%252F%252Flocalhost%253A5000%252Fpreprints%252Fsorarxiv
    ```

## Changes

### Deployment Notes

- [ ] Must be deployed at the same time with https://github.com/CenterForOpenScience/osf.io/pull/6545
- [ ] Must update registered services, example files are uploaded to JIRA

### QA Notes

Please see https://github.com/CenterForOpenScience/osf.io/pull/6545 for QA Notes.

### Example of CAS login Page
![branded-preprints](https://cloud.githubusercontent.com/assets/3750414/20283156/e423964e-aa85-11e6-9792-6d768df33363.png)
![prereg-erpc-preprints](https://cloud.githubusercontent.com/assets/3750414/20283158/e42cf19e-aa85-11e6-8d31-9170ec1834ce.png)

- "Create Account" link now goes to OSF domain with respective path:
  - `/register/?campaign=osf-preprints`
  - `/register/?campaign=socarxiv-preprints`
  - `/register/?campaign=engrxiv-preprints`
  - `/register/?campaign=psyarxiv-preprints`
  - `/register/?campaign=erpc`
  - `/register/?campaign=prereg`

## Side effects
- [x] Talked with product on handle OSF campaign login through institution:
  - To keep the login page simple, if users choose to login through their institution during OSF campaign login, no campaign info is displayed but they will be redirected to where they come from.
